wiring/src/spark_wiring_ble.cpp

```diff
@@ -431,535 +734,577 @@ size_t BleAdvertisingData::locate(const uint8_t* buf, size_t len, BleAdvertising
     return 0;
 }
 
-
 /*******************************************************
  * BleCharacteristicImpl definition
  */
 class BleCharacteristicImpl {
 public:
     BleCharacteristicImpl()
-            : properties(BleCharacteristicProperty::NONE),
-              uuid(),
-              svcUuid(),
-              description(),
-              isLocal(true),
-              dataCb(nullptr),
-              context(nullptr),
-              connHandle(BLE_INVALID_CONN_HANDLE),
-              svcImpl(nullptr),
-              valid_(false) {
+            : isLocal_(false),
+              connHandle_(BLE_INVALID_CONN_HANDLE),
+              properties_(BleCharacteristicProperty::NONE),
+              attrHandles_(),
+              charUuid_(),
+              svcUuid_(),
+              description_(),
+              callback_(nullptr),
+              context_(nullptr) {
     }
 
     BleCharacteristicImpl(const char* desc, BleCharacteristicProperty properties, BleOnDataReceivedCallback callback, void* context)
             : BleCharacteristicImpl() {
-        this->description = desc;
-        this->properties = properties;
-        this->dataCb = callback;
-        this->context = context;
+        properties_ = properties;
+        description_ = desc;
+        callback_ = callback;
+        context_ = context;
     }
 
     BleCharacteristicImpl(const char* desc, BleCharacteristicProperty properties, BleUuid& charUuid, BleUuid& svcUuid, BleOnDataReceivedCallback callback, void* context)
             : BleCharacteristicImpl(desc, properties, callback, context) {
-        this->uuid = charUuid;
-        this->svcUuid = svcUuid;
+        charUuid_ = charUuid;
+        svcUuid_ = svcUuid;
     }
 
     ~BleCharacteristicImpl() = default;
 
-    ssize_t getValue(uint8_t* buf, size_t len) const {
-        if (buf == nullptr || len == 0) {
-            return SYSTEM_ERROR_INVALID_ARGUMENT;
-        }
-        len = len > BLE_MAX_ATTR_VALUE_PACKET_SIZE ? BLE_MAX_ATTR_VALUE_PACKET_SIZE : len;
-        if (isLocal) {
-            return hal_ble_gatt_server_get_characteristic_value(attrHandles.value_handle, buf, len, nullptr);
-        } else if(valid_) {
-            return hal_ble_gatt_client_read(connHandle, attrHandles.value_handle, buf, len, nullptr);
-        } else {
-            return SYSTEM_ERROR_INVALID_STATE;
-        }
+    bool& local() {
```

Comment:
For all the other private members as well? I was thinking that these members are accessed internally only, so simply returning by reference can simplify the class implementation.

Reply:
It's fine to leave as-is, it's just the semantics seem weird at the first glance for such a trivial type (bool). It probably would have been fine to leave the original public member variables instead of getters.